### DISCUSSION TITLE
How good is the resolver part covered by tests?

### DISCUSSION THREAD

**Discussion by laeubi** [eclipse-equinox/equinox#461] [Q&A]
I currently have a problem with the resolver taking very long and not finding a suitable solution (> 20 minutes at very high CPU load) where as a last resort I need to kill the process forcefully.

My idea now is to add some debug output, recompile equinox and probably adjust some places. Of course I don't want to break the resolver and wonder if one runs the test + tck if this gives a good confidence everything "works like before"? Or are there any additional tests to perform (maybe at felix repository?).

**Comment by tjwatson** [comment]
The felix repository has very limited unit tests on resolver.  I find the tests in the `org.eclipse.osgi.tests.container.TestModuleContainer` to be more complete.  In full disclosure, when I make modifications to the felix resolver I do it first in Equinox and added tests there to `TestModuleContainer` and then "contribute" the fix back to felix at my leisure.  I must admit I rarely added tests directly to felix resolver.
